### PR TITLE
Fix: Fetch saved dashboard delivery rule

### PR DESCRIPTION
**Pull request by Xuefeng-Zhu**

### CODE REVIEW

File: packages/dashboards/addon/models/dashboard.js [hunk 1]
```diff
@@ -62,6 +63,19 @@ export default DeliverableItem.extend(Validations, {
     return Ember.A(favoriteDashboards).includes(get(this, 'id'));
   }).volatile(),
 
+  /**
+   * @property {DS.Model} deliveryRuleForUser - delivery rule model
+   */
+  deliveryRuleForUser: computed('user', 'deliveryRules.[]', function () {
```
[the remark this answers is not in the document]
Comment (Xuefeng-Zhu) [reply packages/dashboards/addon/models/dashboard.js]: updated

File: packages/dashboards/addon/models/dashboard.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ import MF from 'model-fragments';
 import DeliverableItem from 'navi-reports/models/deliverable-item';
 import config from 'ember-get-config';
 import { validator, buildValidations } from 'ember-cp-validations';
+import { A as arr } from '@ember/array'
```
Comment (balajimanoharan) [inline packages/dashboards/addon/models/dashboard.js]: Is this needed?

File: packages/reports/addon/models/deliverable-item.js [hunk 3]
```diff
@@ -4,7 +4,22 @@
...
+import { computed, get } from '@ember/object'
 
 export default DS.Model.extend({
-  deliveryRules: DS.hasMany('deliveryRule', { async: true, inverse: 'deliveredItem' })
+  deliveryRules: DS.hasMany('deliveryRule', { async: true, inverse: 'deliveredItem' }),
+
+  /**
+   * @property {DS.Model} deliveryRuleForUser - delivery rule model
+   */
+  deliveryRuleForUser: computed('user', 'deliveryRules.[]', function () {
```
Comment (balajimanoharan) [inline packages/reports/addon/models/deliverable-item.js]: Can we also add a test for this for the dashboard model? Or move the test to the parent model?

**Comment by cythrawll** [comment]
build is failing

**Comment by Xuefeng-Zhu** [comment]
@balajimanoharan I have addressed your comments

File: packages/dashboards/tests/dummy/mirage/fixtures/delivery-rules.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import DeliveryRules from 'navi-dashboards/mirage/fixtures/delivery-rules';
```
Comment (balajimanoharan) [inline packages/dashboards/tests/dummy/mirage/fixtures/delivery-rules.js]: Can we do `export { default } from ....`?

File: packages/dashboards/addon/mirage/fixtures/delivery-rules.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2017, Yahoo Holdings Inc.
```
Comment (balajimanoharan) [inline packages/dashboards/addon/mirage/fixtures/delivery-rules.js]: Can we update the copyright here? `2018`

File: packages/dashboards/tests/unit/models/dashboard-test.js [hunk 6]
```diff
@@ -128,3 +133,23 @@ test('Cloning Dashboards', function(assert) {
...
 });
+
+test('deliveryRuleForUser', function (assert) {
+  assert.expect(1);
+
+  return Ember.run(() => {
+    return Store.findRecord('user', 'navi_user').then(() => {
+      return Store.findRecord('dashboard', 2).then(dashboardModel => {
+        dashboardModel.user = {
+          getUser: () => { return Store.peekRecord('user', 'navi_user'); }
```
Comment (balajimanoharan) [inline packages/dashboards/tests/unit/models/dashboard-test.js]: `getUser: () => Store.peekRecord(...)`

**Pull request merged**